R/parametric-measures.R

```diff
@@ -566,29 +566,29 @@ calc_SMD <- function(A_data, B_data,
 
 # Calculate statistics to be used in calculating log ratio of medians
 
-stats_LRM <- function(data, warn = TRUE) {
+stats_LRM <- function(data, delta_method = FALSE, warn = TRUE) {
   
   n <- length(data)
-  median <- median(data)
-  o1 <- round(n/2 - sqrt(n))
-  if (o1 < 1) {o1 = 1}
+  median_est <- median(data)
+  o1 <- pmax(round(n / 2 - sqrt(n)), 1L)
   o2 <- n - o1 + 1
   p <- pbinom(o1 - 1, size = n, prob = .5)
   z0 <- qnorm(1 - p)
   y <- sort(data)
   L1 <- y[o1]
   U1 <- y[o2]
-  var <- ((U1 - L1) / (2 * z0))^2
-  var_log_delta <- var / (median)^2
-  var_log_bar <- ((log(U1) - log(L1)) / (2 * z0))^2
+  
+  if (delta_method) {
+    var_log_median <- ((U1 - L1) / (2 * z0))^2 / (median_est)^2
+  } else {
+    var_log_median <- ((log(U1) - log(L1)) / (2 * z0))^2  

```

Comment:
It is possible that U1 or L1 is 0, causing var_log_median to be inf. How can we truncate U1 and L1?

Reply:
Yes, which is a problem. Let’s leave it for now and just include a note in the documentation highlighting that this can occur.